exercises/collatz-conjecture/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+// return Ok(x) where x is the number of steps required to reach 1
+pub fn collatz(n: u64) -> Result<u64, &'static str> {
+    if n == 0 {
+        return Err("Only positive numbers are allowed");
```

Comment:
Is it OK to provide this code to the student, rather than simply providing only the function signature and letting the student fill in the entirety of the body? If the code is provided, a disadvantage is that the student doesn't need to do any work (and therefore doesn't get the experience of learning-by-doing) to get the `test_0` to pass.

Reply:
As an early exercise, I think leaving this in is the easiest way to show how Result is used.  The student still needs to figure out how to return Ok().

Reply:
Since I have no knowledge of how to best teach students something, I will simply accept this since I don't know any better. That means my only constraint is: Since this is provided, it must be placed before any other exercise with `Result` (since they won't provide it). If it were removed, there would be more freedom of placement.